Framework/Kernel/inc/MantidKernel/DataItem.h

```diff
@@ -54,6 +54,11 @@ class MANTID_KERNEL_DLL DataItem {
   virtual const std::string toString() const = 0;
   //@}
 
+  /// Acquires a read lock. If another thread currently holds a write lock,
+  /// waits until the write lock is released.
+  void readLock();
```

Comment:
Can we inline this instead as it's just a pass through, i.e.

```cpp
inline void readLock() { getLock()->readLock(); }
```

Reply:
When I do this inline I get `Pointer to incomplete class type is not allowed` on the `getLock()` call. It resolves if I do `include <Poco/RWLock.h>` in this header, but doesn't that mean everything that includes `DataItem.h` will also include this bit of Poco?

Reply:
Ah yes that's a fair point. Maybe it is better to keep this more encapsulated. Let's leave it as you have it 👍🏻